monitoring/iostats_context.cc

```diff
@@ -11,14 +11,19 @@ namespace ROCKSDB_NAMESPACE {
 
 #ifdef ROCKSDB_SUPPORT_THREAD_LOCAL
 __thread IOStatsContext iostats_context;
-#endif
+#else
+#ifndef NIOSTATS_CONTEXT
+#error \
```

Comment:
The old approach seemed effective at only failing the make command if it involved compilation.

Reply:
Yeah, that's right, assuming you are referring to the `#error` approach.

Reply:
It looks like we already use `#error` directives in the codebase. Now, it looks better than the current approach, since we do not have to make these changes to the Makefile and CMakelists.txt